app/models/concerns/thredded/topic_common.rb

```diff
@@ -47,24 +47,45 @@ def unread(user)
           .merge(reads_class.where(reads[:id].eq(nil).or(reads[:read_at].lt(topics[:updated_at]))))
       end
 
+      private
+
+      # @param user [Thredded.user_class]
+      # @return [ByPostableLookup]
+      def read_states_by_topics_lookup(user, null_value)
+        read_states = reflect_on_association(:user_read_states).klass
+          .where(user_id: user.id, postable_id: current_scope.map(&:id))
+        Thredded::TopicCommon::Lookup.new(read_states, null_value)
```

Comment:
@glebm do you really prefer that to the solution that's in the PR at present? If so, I'll go with it, but I  don't see it's much better (it's less clear what's going on and why to my mind).